python/pyiceberg/utils/concurrent.py

```diff
@@ -15,53 +15,7 @@
 # specific language governing permissions and limitations
 # under the License.
 # pylint: disable=redefined-outer-name,arguments-renamed,fixme
-"""Concurrency concepts that support multi-threading."""
-import threading
+"""Concurrency concepts that support efficient multi-threading."""
 from concurrent.futures import Executor, ThreadPoolExecutor
-from contextlib import AbstractContextManager
-from typing import Any, Generic, TypeVar
 
-from typing_extensions import Self
-
-T = TypeVar("T")
-
-
-class Synchronized(Generic[T], AbstractContextManager):  # type: ignore
-    """A context manager that provides concurrency-safe access to a value."""
-
-    value: T
-    lock: threading.Lock
-
-    def __init__(self, value: T, lock: threading.Lock):
-        super().__init__()
-        self.value = value
-        self.lock = lock
-
-    def __enter__(self) -> T:
-        """Acquires a lock, allowing access to the wrapped value."""
-        self.lock.acquire()
-        return self.value
-
-    def __exit__(self, exc_type: Any, exc_value: Any, traceback: Any) -> None:
-        """Releases the lock, allowing other threads to access the value."""
-        self.lock.release()
-
-
-class ManagedExecutor(Executor):
-    """An executor that provides synchronization."""
-
-    def synchronized(self, value: T) -> Synchronized[T]:
-        raise NotImplementedError
-
-
-class ManagedThreadPoolExecutor(ThreadPoolExecutor, ManagedExecutor):
-    """A thread pool executor that provides synchronization."""
-
-    def __enter__(self) -> Self:
-        """Returns the executor itself as a context manager."""
-        super().__enter__()
-        return self
-
-    def synchronized(self, value: T) -> Synchronized[T]:
-        lock = threading.Lock()
-        return Synchronized(value, lock)
+executor: Executor = ThreadPoolExecutor()
```

Comment:
Should we create this one lazily, and able to configure the `max_workers` through the config?